internal/notifications/notifications.go

```diff
@@ -0,0 +1,92 @@
+package notifications
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	"github.com/RHEnVision/provisioning-backend/internal/kafka"
+	"github.com/rs/zerolog"
+)
+
+func init() {
+	GetNotificationClient = getNoopNotificationClient
+}
```

Comment:
I think the problem is here.

You do not need to use init function, in fact, this Go feature should be avoided if possible. It works for database or kafka beacuse implementations are in different packages, but here you have both implementations in the same package.

Can you simply remove init and do this:

func Initialize(ctx context.Context) {
  if config.Application.Notifications.Enabled {
    zerolog.Ctx(ctx).Debug().Msg("Initialized kafka notification client")
    GetNotificationClient = getNotificationClient
  } else {
    zerolog.Ctx(ctx).Debug().Msg("Initialized noop notification client")
    GetNotificationClient = getNoopClient
  }	
}